sale_order_revision/model/sale_order.py

```diff
@@ -64,6 +69,17 @@ def copy_quotation(self):
         msg = _('New revision created: %s') % self.name
         self.message_post(body=msg)
         old_revision.message_post(body=msg)
+
+        # set stored procurement group id on old order
+        old_revision.write({'procurement_group_id': procurement_group_id})
+
+        # swap order lines of old and new order
+        so_line = self.env['sale.order.line']
+        old_lines = so_line.browse(old_revision.order_line.ids)
+        new_lines = so_line.browse(self.order_line.ids)
+        old_lines.write({'order_id': self.id})
+        new_lines.write({'order_id': old_revision.id})
```

Comment:
I tested your shorter version; indeed it goes wrong, the old revision will contain both revision's lines and the new revision contains nothing.